File: src/rust/hello.rs
#!/usr/bin/env rust-script

//! ```cargo
//! [dependencies]
//! clap = { version = "4.5.11", features = ["derive"] }
//! ```

use clap::Parser as _;

#[derive(clap::Parser)]
struct Args {
    #[clap(default_value_t = String::from("Evan"))]
    name: String,
}

fn main() {
    let args = Args::parse();
    println!("Hello, {}!", args.name);
}

File: src/python/hello.py
#!/usr/bin/env -S uv run --quiet

# /// script
# requires-python = ">=3.12"
# dependencies = [
#     "typer-slim",
# ]
# ///

import typer
from typing_extensions import Annotated


app = typer.Typer()


@app.command()
def main(name: Annotated[str, typer.Argument()] = "world"):
    print(f"Hello, {name}!")


if __name__ == "__main__":
    app()
